tools/core-build-tasks/src/tasks/cleanCollateral.ts

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) Mojang AB.  All rights reserved.
+
+import { JsonFile } from "@rushstack/node-core-library";
+import * as fs from "fs";
+import * as Path from "path";
+import rimraf from "rimraf";
+import { getOrThrowFromProcess } from "./helpers/getOrThrowFromProcess";
+
+/**
+ * Format for configuration file.
+ */
+export type CleanCollateralConfigurationFormat = {
+  toClean: string[];
+};
+
+const CLEAN_CONFIG_DEFAULT = "./config/cleanCollateralTask.config.json";
+
+export type CleanCollateralTaskParams = {
+  cleanCollateralTaskConfigPath: string;
+};
+
+/**
+ * Cleans the specified outputs. Outputs could be either folders or files. Has support for the following variable replacements
+ *
+ *   APPDATA, LOCALAPPDATA, PROJECT_NAME
+ *
+ * This constant is replaced at task execution with a value provided by the process environment.
+ *
+ */
+export function cleanCollateralTask(params: CleanCollateralTaskParams) {
```

Comment:
For the purposes of the samples, can this be simplified so that the paths to clean are specified via this param argument, rather than via a config file? The config file is useful for more complex cases, but that's not what we are striving for with this.

Reply:
Done, using a collection in the params object.